modules/core/src/main/java/org/apache/ignite/internal/processors/cache/distributed/dht/preloader/GridDhtPartitionsExchangeFuture.java

```diff
@@ -3489,7 +3489,8 @@ private void resetOwnersByCounter(GridDhtPartitionTopology top,
 
             long maxCntr = maxCntrObj != null ? maxCntrObj.cnt : 0;
 
-            NavigableSet<Long> nonMaxCntrs = e.getValue().headSet(maxCntr, false);
+            NavigableSet<Long> nonMaxCntrs = e.getValue().headSet(maxCntr, false)
+                .tailSet(0L, false);
```

Comment:
Is it a fix for preventing rebalancing reassigned partitions historically?
Let's add an explaining comment.